pkg/scalers/liiklus_scaler.go

```diff
@@ -159,50 +158,17 @@ func (s *liiklusScaler) getLag(ctx context.Context) (uint64, map[uint32]uint64,
 }
 
 func parseLiiklusMetadata(config *scalersconfig.ScalerConfig) (*liiklusMetadata, error) {
-	lagThreshold := defaultLiiklusLagThreshold
-	activationLagThreshold := defaultLiiklusActivationLagThreshold
-
-	if val, ok := config.TriggerMetadata[liiklusLagThresholdMetricName]; ok {
-		t, err := strconv.ParseInt(val, 10, 64)
-		if err != nil {
-			return nil, fmt.Errorf("error parsing %s: %w", liiklusLagThresholdMetricName, err)
-		}
-		lagThreshold = t
+	meta := &liiklusMetadata{}
+	if err := config.TypedConfig(meta); err != nil {
+		return nil, fmt.Errorf("error parsing liiklus metadata: %w", err)
 	}
-
-	if val, ok := config.TriggerMetadata[liiklusActivationLagThresholdMetricName]; ok {
-		t, err := strconv.ParseInt(val, 10, 64)
-		if err != nil {
-			return nil, fmt.Errorf("error parsing %s: %w", liiklusActivationLagThresholdMetricName, err)
-		}
-		activationLagThreshold = t
-	}
-
-	groupVersion := uint32(0)
-	if val, ok := config.TriggerMetadata["groupVersion"]; ok {
-		t, err := strconv.ParseUint(val, 10, 32)
-		if err != nil {
-			return nil, fmt.Errorf("error parsing groupVersion: %w", err)
-		}
-		groupVersion = uint32(t)
-	}
-
 	switch {
-	case config.TriggerMetadata["topic"] == "":
+	case meta.Topic == "":
 		return nil, ErrLiiklusNoTopic
-	case config.TriggerMetadata["address"] == "":
+	case meta.Address == "":
 		return nil, ErrLiiklusNoAddress
-	case config.TriggerMetadata["group"] == "":
+	case meta.Group == "":
 		return nil, ErrLiiklusNoGroup
 	}
-
-	return &liiklusMetadata{
-		topic:                  config.TriggerMetadata["topic"],
-		address:                config.TriggerMetadata["address"],
-		group:                  config.TriggerMetadata["group"],
-		groupVersion:           groupVersion,
-		lagThreshold:           lagThreshold,
-		activationLagThreshold: activationLagThreshold,
-		triggerIndex:           config.TriggerIndex,
-	}, nil
+	return meta, nil
```

Comment:
I think this is the place that should populate `meta.triggerIndex`
```suggestion
    meta.triggerIndex = config.TriggerIndex
	return meta, nil
```

Reply:
Added in: [4934fd0](https://github.com/kedacore/keda/pull/6433/commits/4934fd095132403bbef5a4e781424fa6985a0850)